modules/fee-market/src/s2s/payment.rs

```diff
@@ -202,29 +193,26 @@ where
 					}
 					total_slash += assigned_relayers_slash;
 
-					// 80% total slash => confirm relayer
+					// MessageRelayersRewardRatio total slash => message relayer
 					message_reward = T::MessageRelayersRewardRatio::get() * total_slash;
-					// 20% total slash => confirm relayer
+					// ConfirmRelayersRewardRatio total slash => confirm relayer
 					confirm_reward = T::ConfirmRelayersRewardRatio::get() * total_slash;
 				}
 
-				// Update confirmation relayer total rewards
-				confirmation_rewards = confirmation_rewards.saturating_add(confirm_reward);
-				// Update message relayers total rewards
-				messages_rewards
-					.entry(entry.relayer.clone())
-					.and_modify(|r| *r = r.saturating_add(message_reward))
-					.or_insert(message_reward);
+				reward_item.to_message_relayer = Some((entry.clone().relayer, message_reward));
+				reward_item.to_confirm_relayer = Some((confirm_relayer.clone(), confirm_reward));
+
+				Pallet::<T, I>::deposit_event(Event::OrderReward(
+					lane_id,
+					message_nonce,
+					reward_item.clone(),
+				));
```

Comment:
I'm not sure whether we should deposit event in this function(slash_and_calculate_rewards) or at the end of pay_relayers_rewards.

Reply:
If deposit this event at the end of `pay_relayers_rewards`, you may need to remove the `RewardBook` and iterate the whole `RewardItems`. I prefer the current way.